src/win.rs

```diff
@@ -117,26 +118,37 @@ impl Clone for HandleRef {
     }
 }
 
+/// Ensures the handle is never null by converting null values to the
+/// pseudo-handle `INVALID_HANDLE_VALUE`.
+fn nonnull_handle<T: AsRawHandle>(stdio: T) -> RawHandle {
+    let handle = stdio.as_raw_handle();
+    if handle.is_null() {
+        INVALID_HANDLE_VALUE
+    } else {
+        handle
+    }
+}
+
 impl HandleRef {
     /// Create a borrowed handle to stdin.
     ///
     /// When the returned handle is dropped, stdin is not closed.
     pub fn stdin() -> HandleRef {
-        unsafe { HandleRef::from_raw_handle(io::stdin().as_raw_handle()) }
+        unsafe { HandleRef::from_raw_handle(nonnull_handle(io::stdin())) }
```

Comment:
Can we include a note in the stdin/stdout/stderr docs here that say what the behavior here is when stdin/stdout/stderr aren't available? I think we should also specifically mention the Windows GUI subsystem thing, to make it particularly concrete.

Reply:
I've added some documentation. I hedged slightly, saying using such a handle is "likely" to fail because I'm not sure I can guarantee the behaviour of every file function in the Windows API, now and in the future.